### PR TITLE
Fe add folders

### PR DESCRIPTION
**Pull request by tomimarkus991**
1. Updated folders
2. Added more strict ts config
3. removed react import as it is not needed after V17 on React
4. Fixed eslint devdependency error
5. added dev script

### CODE REVIEW

File: fe/src/pages/About/About.tsx [hunk 1]
```diff
@@ -1,5 +1,3 @@
-import React from 'react';
```
Comment (GeorgDV) [inline fe/src/pages/About/About.tsx:1]: I'm not sure about removing `react` imports, altough true, that its not required for react to work.
IMHO this rises more problems with eslint than fixes, so i'd say we should add the imports back. 🤔 
errors

**Review by GeorgDV — CHANGES_REQUESTED** [review]

File: fe/src/pages/About/About.tsx (hunk 1 above)
Comment (GeorgDV) [reply fe/src/pages/About/About.tsx:1]: Actually when i restarted local project the errors were gone, so the `react/react-in-jsx-scope`  rule works after all.

I'd say then we can leave the imports as they now are.

File: fe/src/App.tsx [hunk 2]
```diff
@@ -1,8 +1,7 @@
-import React from 'react';
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
-import Home from './components/pages/Home';
-import About from './components/pages/About/index';
-import Error from './components/pages/Error/index';
+import Home from './pages/Home';
+import About from './pages/About/index';
+import Error from './pages/Error/index';
```
[the remark this answers is not in the document]
Comment (Kasparsu) [reply fe/src/App.tsx:4]: just to note that both are okay structures and I proposed both of them. However it was allready agreed that we are going with the structure @GeorgDV mentioned above.

**Comment by Kasparsu** [comment]
Before merging fixup commits. commits that have sole purpose of fixing your past mistakes should not end up in main. you can do interactive rebase and then squash or fixup commits and even reorder them if need be. https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#interactive-rebase

File: fe/src/App.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomimarkus991) [reply fe/src/App.tsx:4]: Having a specific folder for route-level components to be placed is a good practice. It also sets a clear hierarchy of differentiation between pages and components, where a page can import a component to display it and do nothing else, not even Business Logic.

You can also read [this](https://www.freecodecamp.org/news/a-better-way-to-structure-react-projects/)

[the remark this answers is not in the document]
Comment (GeorgDV) [reply fe/src/App.tsx:4]: I guess both of them can be useful structures, but it is already planned upon current structure, as said. In current hierarchy there is still a set difference between page components and other components, but its just differently structured.

[the remark this answers is not in the document]
Comment (Kasparsu) [reply fe/src/App.tsx:4]: If I think about it a while. this change actually makes bit more sense. I would suggest we change the structure proposed here. @GeorgDV change is quite small and it works either way. But having pages and layouts folders and then bunch of loose component folders all mixed up might actually be bad idea. So maybe yeah leave components folder for loose components. either reusable or single use but pages and layouts are their own separate things.

[the remark this answers is not in the document]
Comment (Kasparsu) [reply fe/src/App.tsx:4]: maybe can get @Orzelius opinion here?

[the remark this answers is not in the document]
Comment (Orzelius) [reply fe/src/App.tsx:4]: This has long deviated from the structure I'm familiar with so I can't really say. Usually in react everything is a component and the topmost ones might happen to be pages as well.

In the article Tomi linked, an indea of multiple contexts was brought up, which I'm not really familiar with. Usually there's one global context and then further state shared between components. 

But all in all I think we can go with this approach (proposed in this pr). Seems like it's a bit more React-friendly and requires less inventing.

**Review by Kasparsu — APPROVED** [review]

File: fe/src/App.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kasparsu) [reply fe/src/App.tsx:4]: @GeorgDV maybe then resolve and merge?

[the remark this answers is not in the document]
Comment (GeorgDV) [reply fe/src/App.tsx:4]: Alrighty then, we'll go with this structure. 👍

**Review by GeorgDV — APPROVED** [review]

**Comment by GeorgDV** [comment]
@tomimarkus991 This branch has conflicts so they need fixing before merging.

Also with the new structure could you please update that [this](https://github.com/TPT-Loane/TPT-Loane/pull/31) added eslint rule works with the new directory structure?
Basically you only need changes [here ](https://github.com/TPT-Loane/TPT-Loane/pull/31/files#diff-ace789a57d4e936793d1f602a82b8210450c3f0470a940391084a527bfdda75fR39)in "allow" field to correspond with new directories, so that imports like **'./pages/Home**' are **allowed** (./pages/*), etc.

**Comment by tomimarkus991** [comment]
> 
> 
> @tomimarkus991 This branch has conflicts so they should be fixed and rebased.
> 
> Also with the new structure could you please update that [this](https://github.com/TPT-Loane/TPT-Loane/pull/31) added eslint rule works with the new directory structure?
> Basically you only need changes [here ](https://github.com/TPT-Loane/TPT-Loane/pull/31/files#diff-ace789a57d4e936793d1f602a82b8210450c3f0470a940391084a527bfdda75fR39)in "allow" field to correspond with new directories, so that imports like **'./pages/Home**' are **allowed** (./pages/*), etc.

will do

**Pull request merged**